service-base/src/test/java/org/eclipse/hono/service/tenant/TenantAmqpEndpointTest.java

```diff
@@ -0,0 +1,97 @@
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 1.0 which is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ */
+
+package org.eclipse.hono.service.tenant;
+
+import static org.eclipse.hono.util.TenantConstants.Action;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.contains;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import org.apache.qpid.proton.message.Message;
+import org.eclipse.hono.util.*;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.eventbus.EventBus;
+import io.vertx.core.json.JsonObject;
+import io.vertx.proton.ProtonHelper;
+
+/**
+ * Tests verifying behavior of {@link TenantAmqpEndpoint}.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class TenantAmqpEndpointTest {
+
+    private static final ResourceIdentifier resource = ResourceIdentifier.from(TenantConstants.TENANT_ENDPOINT, Constants.DEFAULT_TENANT, null);
+
+    @Mock private EventBus eventBus;
+    @Mock private Vertx    vertx;
+
+    private TenantAmqpEndpoint endpoint;
+
+    /**
+     * Sets up the fixture.
+     */
+    @Before
+    public void setUp() {
+
+        when(vertx.eventBus()).thenReturn(eventBus);
+
+        endpoint = new TenantAmqpEndpoint(vertx);
+    }
+
+    /**
+     * Verifies that the endpoint forwards a request message via the event bus.
+     */
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testProcessMessageSendsRequestViaEventBus() {
+
+        Message msg = ProtonHelper.message();
+        msg.setSubject(Action.ACTION_GET.toString());
+        MessageHelper.addTenantId(msg, Constants.DEFAULT_TENANT);
+        MessageHelper.annotate(msg, resource);
+
+        endpoint.processRequest(msg, resource, Constants.PRINCIPAL_ANONYMOUS);
+
+        verify(eventBus).send(contains(TenantConstants.EVENT_BUS_ADDRESS_TENANT_IN), any(JsonObject.class), any(Handler.class));
+    }
+
+    /**
+     * Verifies that the JsonObject constructed for a tenant message on the event bus contains the tenantId
+     * as defined in the {@link RequestResponseApiConstants} class.
+     */
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testTenantMessageForEventBus() {
+
+        Message msg = ProtonHelper.message();
+        msg.setSubject(Action.ACTION_GET.toString());
+        MessageHelper.addTenantId(msg, Constants.DEFAULT_TENANT);
+        MessageHelper.annotate(msg, resource);
+
+        final JsonObject tenantMsg = TenantConstants.getTenantMsg(msg);
```

Comment:
this seems to be a unit test for class `TenantConstants`

Reply:
Correct - will move it. `CredentialsAmqEndpointTest` is also affected.